doc/rst/technotes/foreach.rst

```diff
@@ -0,0 +1,49 @@
+.. _readme-foreach:
+
+====================
+The `foreach` Loop
+====================
+
+A ``foreach`` loop signifies that loop iterations are order-independent. This
+enables the compiler to vectorize the loop or to offload its execution to the
+GPU. In contrast, a ``for`` loop is order-dependent, and a ``forall`` loop can
+create parallel tasks and distribute iterations based on the iterator. The tasks
+created by ``forall`` can execute loop iterations in order-independent manner
+*if* the iterator uses ``foreach``.
```

Comment:
Both `foreach` and `forall` indicate that *the current loop* is order independent. This document describes later how this information is merged when using an iterator. I'd like this lead-in paragraph to focus more on the idea of order independence of the current loop and to leave the detail of what the iterator has to do with it for another paragraph.

Reply:
Ah, I see what you mean better now. Good point.